pkg/controller/taintmanager/OWNERS

```diff
@@ -0,0 +1,12 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
+  - sig-node-approvers
+reviewers:
+  - sig-node-reviewers
+emeritus_approvers:
+  - bowei
+  - k82cn
+  - gmarek
+labels:
+  - sig/node
```

Comment:
This is copied from `pkg/controller/nodelifecycle`. We should discuss as part of this PR the ownership for the component now that has been de-coupled from the lifecycle controller.

Reply:
Per https://github.com/kubernetes/kubernetes/issues/116771#issuecomment-1499393771, sig-scheduling should be owning this folder.

Reply:
Changed this in a2f34ba4a9a